src/app/epics/get-items.epic.ts

```diff
@@ -23,19 +24,21 @@ export class GetItemsEpic {
         private searchHistoryActions: SearchHistoryActions,
     ) {}
 
-    getDefaultItems = (action$): Observable<AnyAction> => {
+    getDefaultItems = (action$: ActionsObservable<AnyAction>): Observable<AnyAction> => {
         return action$.ofType(ItemsActions.GET_DEFAULT_ITEMS)
             .do( (action: AnyAction) => console.log('Epic sees action: ', action.type) )
             .mergeMap( () => this.appService.getTrendingItems() )
             .map((items:IItemsRef) => this.itemsActions.recievedDefaultItems(items, Date.now()) );
     }
 
-    getItemsByQuery = (action$): Observable<AnyAction> => {
+    getItemsByQuery = (action$: ActionsObservable<AnyAction>): Observable<AnyAction> => {
         const queryAndItems$: Observable<[string, IItemsRef]> = action$
             .ofType(ItemsActions.GET_ITEMS_BY_QUERY)
             .map( (action: AnyAction) => action.query )
             .switchMap( (query: string) => 
-                this.appService.getItemsByQuery(query).map( (items: IItemsRef) => [query, items] ) 
+                this.appService.getItemsByQuery(query).map( (items: IItemsRef) => 
+                    [query, items] as [string, IItemsRef] 
```

Comment:
Is casting here necessary? I would expect typescript already knows `query` is a `string`, and you've already typed `items` as  `IItemsRef`